scripts/lint-style.py

```diff
@@ -295,10 +303,16 @@ def format_errors(errors):
             output_message(path, line_nr, "ERR_DOT", "Line is an isolated focusing dot or uses . instead of ·")
         if errno == ERR_SEM:
             output_message(path, line_nr, "ERR_SEM", "Line contains a space before a semicolon")
+        if errno == ERR_WIN:
+            output_message(path, line_nr, "ERR_WIN", "Windows line endings (\\r\\n) detected")
 
 def lint(path):
-    with path.open(encoding="utf-8") as f:
+    with path.open("rb") as f:
         lines = f.readlines()
+        errs = windows_line_endings_check(lines, path)
+        format_errors(errs)
+        lines = [line.decode("utf-8").rstrip() + "\n" for line in lines]
+
```

Comment:
We still need to replace `\r\n` with `\n` here before performing other checks. Otherwise, `import_only_check`will raise an `IndexError` when there is an empty line `\r\n`.
https://github.com/leanprover-community/mathlib4/blob/ba80034619884735df10ebb449b4fc2a44b2c3e7/scripts/lint-style.py#L166

An alternative is to add `\r\n` to this line:
https://github.com/leanprover-community/mathlib4/blob/ba80034619884735df10ebb449b4fc2a44b2c3e7/scripts/lint-style.py#L95C17-L95C17

Reply:
Ah, then I'm fine with keeping it.